### PR TITLE
Chore: restrict origin with cors and secure app with helmet

### PR DESCRIPTION
**Pull request by twumm**
# Description
Install and use `cors` and `helmet` on the server.
For `cors`, options have been passed in to limit the requests that come in based on the url allowed.

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Change status
- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?
Requests were made to the server from external app - `postwoman.io` - this failed because `postwoman.io` is not allowed by our `cors` whitelist.
Requests were made again with `postwoman.io` after adding it to our whitelist. This passed as expected.

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if relevant
- [x] I have run the app using my feature and ensured that no functionality is broken
- [ ] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by twumm** [comment]
[Restrict requests to the API](https://trello.com/c/h8lyAEfZ/50-restrict-requests-to-the-api)

File: utils/middlewares.js [hunk 1]
```diff
@@ -1,8 +1,24 @@
 const morgan = require("morgan");
 const express = require("express");
+const cors = require("cors");
+const helmet = require("helmet");
 
-module.exports = function(app) {
+const CORS_ORIGIN_WHITELIST = ['http://localhost:3000'];
```
Comment (abidex4yemi) [inline utils/middlewares.js:6]: We need to add hosted front-end `URL` to the white list

Comment (abidex4yemi) [reply utils/middlewares.js:6]: We should also check for the environment for `dev` or `prod` in order to know which options to use

**Review by abidex4yemi — CHANGES_REQUESTED** [review]
Good job on this one, Left some comment below

File: utils/middlewares.js (hunk 1 above)
Comment (twumm) [reply utils/middlewares.js:6]: Yes, the front-end url will be added.

For the check of the environment, I do not think doing that will add any extra ability to this method.

**Pull request merged**